fix(agent): Fix agent.secure.enabled flag

aroberts87 opened the pull request:
## What this PR does / why we need it:
Setting `agent.secure.enabled=false` was not explicitly disabling the secure features in the generated agent configuration. Instead, the behavior was left up to the agent process's defaults, or a backend config push. This behavior is not desirable as a user who explicitly set `agent.secure.enabled=false` could have ended up with secure components running anyway. This fix ensures that setting `agent.secure.enabled=false` explicitly generates the configuration entries to force all secure components to be disabled.

## Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [x] Title of the PR starts with type and scope, (e.g. feat(agent,node-analyzer,sysdig-deploy):)
- [x] Chart Version bumped for the respective charts
- [x] Check GithubAction checks (like lint) to avoid merge-check stoppers
- [x] All test files are added in the tests folder of their respective chart and have a "_test" suffix

Check Contribution guidelines in README.md for more insight.

aroberts87 committed:
Fix agent.secure.enabled flag

Setting agent.secure.enabled=false was not explicitly disabling
the secure features in the generated agent configuration.
Instead, the behavior was left up to the agent process's
defaults, or a backend config push. This behavior is not
desirable as a user who explicitly set agent.secure.enabled=false
could have ended up with secure components running anyway.
This fix ensures that setting agent.secure.enabled=false
explicitly generates the configuration entries to force
all secure components to be disabled.

lilx1ao commented on charts/agent/templates/configmap.yaml:
```diff
@@ -13,20 +13,10 @@ data:
-      enabled: true
-  {{- end }}
+{{- if and (not .Values.secure.enabled) .Values.auditLog.enabled }}
+  {{- fail "auditLog.enabled cannot be true when secure.enabled is false" }}
```
tiny suggestion: might make sense to do it from within the agent.secureFeatures?

aroberts87 replied:
Sure, I can do this. I think I tried that before and ended up with weird blank lines that I couldn't solve with adding `-`'s, but I agree with you it would look cleaner in the template. Let me try again. 💪🏻

lilx1ao replied:
💪

aroberts87 replied:
Done!

aroberts87 committed:
Update default value of agent.auditLog.enabled

The existing default value of false was not actually
rendered by the charts in any way which ended up meaning
that whatever the Agent's default value was would be what
ended up running. An earlier commit fixed the helm template
logic where secure.enabled and auditLog.enabled being false
were essentially no-ops and now actually explicitly disables
them. After that change, having auditLog.enabled being false
in the values.yaml file created a break in default behavior
as deploying the charts with the default values would result
in k8s_audit_server being disabled now versus enabled as it
was before. Since the default value in the Agent itself is
true, the helm charts are now being adjusted to reflect that.

balinaveen commented on charts/agent/templates/_helpers.tpl:
```diff
@@ -318,3 +318,31 @@ Use global sysdig tags for agent
+*/}}
+{{- define "agent.secureFeatures" }}
+    {{- $secureBlockConfig := dict "security" (dict
+        "enabled" .Values.secure.enabled
```
May want to explain why secure.enabled = true and false are not mirror images.
For secure.enabled, only security.enabled is set to true (along with the k8s_audit_server_enabled set to whatever the auditLog.enabled is)
For secure.enabled=false, all known secure features are disabled.

The explanation is that for secure.enabled=true we depend on the backend to enable the appropriate secure features for the customer. And since we don't know a priori what secure features may be enabled when secure.enabled=true we disable them all when secure.enabled=false

aroberts87 replied:
Done!

balinaveen replied:
Thanks

aroberts87 committed:
Explain why secure.enabled true/false behaves differently

aroberts87 committed:
Bump agent chart version
